shm/Game.cpp

```diff
@@ -9,4 +12,37 @@ bool Game::checkWinCondition() const {
 
 bool Game::checkLoseCondition() const {
     return (player_->getMoney() == 0);
+}
+
+void Game::Travel() {
+    std::cout << *map_; //unimplemented << operator for map
+    auto destination = map_->getIsland(getTravelLocation());
+    if(!destination) {
+        return;
+    }
+
+    auto distance = map_->getDistanceToIsland(destination);
+    map_->setCurrentPosition(destination);
+    advanceTimeTraveling(distance);
+
+}
+
+Coordinates Game::getTravelLocation() {
+    std::cout << "Type position X of an Island to travel to: "; //we assume that << operator will give Islands coords
+    size_t X{};
+    std::cin >> X;
+    std::cout << "\nType position Y of an Island to travel to: ";
+    size_t Y{};
+    std::cin >> Y;
+    std::cout <<"\n";
+
+    return Coordinates(X, Y);
```

Comment:
Dont we need some sort of validation?

Reply:
It's not a job for this function. If these coordinates return nullptr then Travel just ends executing. it could be modified after we learn exceptions

Reply:
Could you explain what happen in this code? The player can choose the island's coordinates by himself? What about the islands that we generated before?:)

Reply:
Map with Islands coordinates get's printed, User chooses an island to go to and enters its coordinates